code/modules/admin/verbs/randomverbs.dm

```diff
@@ -104,11 +104,44 @@
 	else
 		to_chat(usr, SPAN_WARNING("You need to be a ghost in order to use this."))
 
-// bite me
-var/global/explosive_antigrief_on = TRUE
-/client/proc/toggle_explosive_antigrief()
-	set name = "Toggle Explosive Antigrief"
+/client/proc/set_explosive_antigrief()
+	set name = "Set Explosive Antigrief"
+	set category = "Admin.Game"
+
+	if(!admin_holder || !(admin_holder.rights & R_MOD))
+		return
+
+	var/antigrief_choice = tgui_input_list(usr, "Select the preferred antigrief type:", "Select", list("Enabled", "Enabled for new players", "Disabled", "Cancel"))
+	if(!antigrief_choice || antigrief_choice == "Cancel")
+		return
+
+	switch(antigrief_choice)
+		if("Disabled")
+			CONFIG_SET(number/explosive_antigrief, ANTIGRIEF_DISABLED)
+			message_staff(FONT_SIZE_LARGE("[key_name_admin(usr)] has disabled explosive antigrief."))
+		if("Enabled")
+			message_staff(FONT_SIZE_LARGE("[key_name_admin(usr)] has fully enabled explosive antigrief for all players."))
+			CONFIG_SET(number/explosive_antigrief, ANTIGRIEF_ENABLED)
+		if("Enabled for new players")
```

Comment:
```suggestion
#define ANTIGRIEF_OPTION_ENABLED "Enabled"
#define ANTIGRIEF_OPTION_NEW_PLAYERS "Enabled for New Players"
#define ANTIGRIEF_OPTION_DISABLED "Disabled"

	var/antigrief_choice = tgui_input_list(usr, "Select the preferred antigrief type:", "Select", list(ANTIGRIEF_OPTION_ENABLED, ANTIGRIEF_OPTION_NEW_PLAYERS, ANTIGRIEF_OPTION_DISABLED))
	if(!antigrief_choice)
		return

	switch(antigrief_choice)
		if(ANTIGRIEF_OPTION_DISABLED)
			CONFIG_SET(number/explosive_antigrief, ANTIGRIEF_DISABLED)
			message_staff(FONT_SIZE_LARGE("[key_name_admin(usr)] has disabled explosive antigrief."))
		if(ANTIGRIEF_OPTION_ENABLED)
			message_staff(FONT_SIZE_LARGE("[key_name_admin(usr)] has fully enabled explosive antigrief for all players."))
			CONFIG_SET(number/explosive_antigrief, ANTIGRIEF_ENABLED)
		if(ANTIGRIEF_OPTION_NEW_PLAYERS)
		
#undef ANTIGRIEF_OPTION_ENABLED
#undef ANTIGRIEF_OPTION_NEW_PLAYERS
#undef ANTIGRIEF_OPTION_DISABLED
```

move the define and undefine appropriately, though

Reply:
move them where